voila/handler.py

```diff
@@ -127,7 +127,10 @@ async def get_generator(self, path=None):
                 kernel_spec_manager=self.kernel_spec_manager,
             )
 
-            await gen.initialize(template=template_arg, theme=theme_arg)
+            done = await gen.initialize(template=template_arg, theme=theme_arg)
+            if not done:
+                self.redirect_to_file(path)
```

Comment:
I am not sure I understand the PR, if the `path` was not a path to a Notebook file in the first place, maybe we should not use the `NotebookRenderer` class?

Reply:
Indeed, it makes no sense to access non-notebook files through `/voila/render/` URL. But in some corner cases, where users have `IFrame` or `FileLink` widgets in their notebooks, these widgets will detect the URL to the local files from the current URL of the notebook, which is `/voila/render/...` 
So this PR will redirect these requests to the normal `/voila/files/`

Reply:
I see. Could we detect this earlier in the code? Instead of trying to initialize a `NotebookRenderer` generator, we could probably redirect to the FileHandler somewhere around [here](https://github.com/voila-dashboards/voila/pull/1073/files#diff-7bc454003168bbff49060f6b1bef6e8d149654b85c18edbf5ca75f1720fdeeffR40)? 

I guess if the path does not end with `.ipynb` we are safe to try to redirect?

Reply:
`Voila` can run text files by setting the mapping `VoilaConfiguration.extension_language_mapping`, so the file extension check would be more complicated.

Reply:
Ok I see, thanks!